Created a check that determines if the CSV has the minimum required info

zzamn opened the pull request:
Running the csvCheck() function specifies where the problematic lines are in the CSV file. It returns a boolean.

Option to return false immediately even if one line is wrong, but its been commented out. So right now it goes through every line, and prints which lines are problematic.

Hope this is alright

zzamn committed:
Created a check that determines if the CSV has the minimum required info

It specifies where the problematic lines are in the CSV file.

Option to return false immediately even if one line is wrong.

But its been commented out

zzamn commented:
NEVER MIND. I DIDN'T DO A CHECK TO SEE IF THE EMAIL IS IN THE RIGHT FORMAT

zzamn commented:
2 MINUTES

CloudArtAdmins commented:
Haha, we are in no rush here.
This is for long term usage, so no worries.

CloudArtAdmins committed:
Test email should be a seagl email

zzamn committed:
Added a rudimentary mail validity checker

zzamn committed:
Merge branch 'main' of https://github.com/zzamn/templated_emails

zzamn committed:
Added is_invalid_email, modified csv_check

checks if email is of the form x@seagl.org

Alternate option which checks if email is of form x@seagl.y has been commented out

also, slightly modified csv_check so that it doesn't only catch empty strings, but also strings ONLY composed of whitespaces

zzamn commented:
I've added a function is_invalid_email. Which checks if every mail is of the format x@seagl.org . 
If you actually wanted it to be of the format x@seagl.y , where y can be 'org' or 'com' or anything else, that's also in the code. I just commented it out. 

the csv_check function checks if the email is valid using is_invalid email and makes sure that the organization and contact name have at least one character. So, they can't be empty or have ONLY blank spaces.

zzamn commented:
This is based on the assumption that the Organization, Contact_Name and Contact_Email are the only essential requirements, and Cascadia and Prev_Sponsor are optional.

If that isn't the case, I can easily modify it. 

Now that I think about it, I should have asked for clarifications super early on. Sorry if what I wrote wasn't what you were looking for 😅

CloudArtAdmins commented:
You are correct about the required and not fields. So no problem there. If you made them "required" this would be no problem either, anyway.

CloudArtAdmins commented:
Okay, so reviewed it.
We actually need this to just check that an email is in the email field. We do not want to specify that it is a SeaGL domain, because this is for template emails to partners, who are highly unlikely to have a SeaGL email.
Could you modify the email check to not care what the domain is, please?

I think the rest is good, though. So once done, I'll pull this version and test. 
If it works, it will be merged.

zzamn committed:
is_invalid_email checks if input is email

Now, it doesn't check whether the email is of the seagl format,
But rather if it is in an email format (i.e x@y.z)

zzamn commented:
I updated the is_invalid_email function. 

It now checks if email is of the format ' x@y.z ' where:
        - len(x), len(y)>0
        - len(z)>1
        - x can consist of alphanumeric characters ,   '.'   ,   '_'   ,   '%'   ,    '+'  and  '-' 
        - y can consist of alphanumeric characters,   '.' and  '-' 
        - z can consist of ONLY alphanumeric characters
